### PR TITLE
Add DMX block addressing and a test

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: firmware/src/dimmer_model.c [hunk 1]
```diff
@@ -72,14 +110,73 @@ int DimmerModel_SetIdentifyMode(const RDMHeader *header,
...
+  bool is_contiguous = true;
+  unsigned int i = 0;
+  for (; i < NUMBER_OF_SUB_DEVICES; i++) {
+    RDMResponder *responder = &g_subdevices[i].responder;
+    uint16_t sub_device_footprint = responder->def
+        ->personalities[responder->current_personality - 1].slot_count;
+    total_footprint += sub_device_footprint;
+    if (expected_start_address &&
+        expected_start_address != responder->dmx_start_address) {
+        is_contiguous = false;
```
Comment (peternewman) [inline firmware/src/dimmer_model.c]: De-dent by 2

File: firmware/src/dimmer_model.c [hunk 2]
```diff
@@ -72,14 +110,73 @@ int DimmerModel_SetIdentifyMode(const RDMHeader *header,
...
-void DimmerModel_Initialize(const DimmerModelSettings *settings) {
+void DimmerModel_Initialize() {
   RDMResponder *temp = g_responder;
 
   unsigned int i = 0;
+  uint16_t sub_device_index = 1;
   for (; i < NUMBER_OF_SUB_DEVICES; i++) {
-    g_subdevices[i].index = i + 1;  // subdevices start from 1
+    if (i == 1) {
+      // Leave at gap at sub-device 2, since sub devices aren't required to be
```
Comment (peternewman) [inline firmware/src/dimmer_model.c]: SPaG a gap at

**Comment by peternewman** [comment]
LGTM, just minor comments.

File: firmware/src/dimmer_model.c (hunk 2 above)
Comment (nomis52) [inline firmware/src/dimmer_model.c]: Fixed

File: firmware/src/dimmer_model.c (hunk 1 above)
Comment (nomis52) [inline firmware/src/dimmer_model.c]: Done

**Comment by peternewman** [comment]
LGTM

**Pull request merged**